052-cruise-control-api-users.md

```diff
@@ -0,0 +1,274 @@
+
+Cruise Control's authorization credential files adhere to [Jetty's HashLoginService's file format](https://eclipse.dev/jetty/javadoc/jetty-10/org/eclipse/jetty/security/HashLoginService.html): `username: password [,rolename ...]`.
+
+By default Cruise Control defines three roles: `VIEWER`, `USER` and `ADMIN`.
+```
+VIEWER role: has access to the most lightweight kafka_cluster_state, user_tasks and review_board endpoints.
+USER role: has access to all the GET endpoints except bootstrap and train.
+ADMIN role: has access to all endpoints.
+```
+This proposal is for supporting the `USER` and `VIEWER` roles only.
+For more details on these roles checkout the [Cruise Control Wiki](https://github.com/linkedin/cruise-control/wiki/Security#authorization)
+
+So a developer could create a Cruise Control API authentication credentials file named `cruise-control-auth.txt`, with the following content:
+```
+userOne: passwordOne, USER
+userTwo: passwordTwo, VIEWER
+```
+
+Then the developer can use this file to create a secret using the following command:
+```
+kubectl create secret generic cruise-control-api-users-secret  --from-file=key=cruise-control-auth.txt
+```
+
+The command would create a secret:
+```yaml=
+apiVersion: v1
+kind: Secret
+metadata:
+  name: cruise-control-api-users-secret
+type: Opaque
+data:
+  key: dXNlck9uZTogcGFzc3dvcmRPbmUsIFVTRVIKdXNlclR3bzogcGFzc3dvcmRUd28sIFZJRVdFUgo=
+```
+
+The developer could then reference the secret in a `spec.cruiseControl.apiUsers` section of the `Kafka` resource.
+
+The schema would look like this:
+
+```yaml=
+apiVersion: kafka.strimzi.io/v1beta2
+kind: Kafka
+metadata:
+  ...
+spec:
+  cruiseControl:
+    apiUsers:
+      type: hashloginservice (1)
+      valueFrom: (2)
+        secretKeyRef:
+          name: cruise-control-api-users-secret
+          key: key
+     ...
+```
+(1) A `type` field is added here to describe the format of the data and for configuration flexibility in the future. 
+This gives us the option to add different data format types in the future. 
+In this example we use Jetty's HashLoginService format since that is the format which Cruise Control uses for its API user configuration.
+
+(2) The `valueFrom` construct allows us to add more sources in the future if needed. 
+This is also a pattern used in other Strimzi APIs already, for example Strimzi's logging configuration, password configuration in KafkaUser resources, metrics configuration, and more.
+
+Strimzi would decode and use the contents of this secret to populate the Cruise Control API auth credentials file which would be mounted and used by the Cruise Control pod.
+
+It's worth noting that if a developer were to provide a corrupt configuration file - one which specifies the `ADMIN` role or an incorrect format, the Strimzi Operator would ignore the custom configuration, log a warning, and add a warning condition to the `Kafka` custom resource.
```

Comment:
Can you show an example snippet of YAML, so as ot be explicit about what the API is?

Reply:
@tombentley  Just added an example, let me know what you think!